api/v1alpha1/nicclusterpolicy_types.go

```diff
@@ -110,6 +110,41 @@ type PSPSpec struct {
 	Enabled bool `json:"enabled,omitempty"`
 }
 
+// DrainSpec describes configuration for node drain during automatic upgrade
+type DrainSpec struct {
+	// Enable indicates if node draining is allowed during upgrade
+	// +optional
+	// +kubebuilder:default:=true
+	Enable bool `json:"enable,omitempty"`
+	// Force indicates if force draining is allowed
+	// +optional
+	// +kubebuilder:default:=false
+	Force bool `json:"force,omitempty"`
+	// +optional
+	PodSelector string `json:"podSelector,omitempty"`
```

Comment:
@ykulazhenkov `drain.Helper` structure that is used in drain API, receives `PodSelector` as a single string. https://pkg.go.dev/k8s.io/kubectl/pkg/drain#Helper

Reply:
> @ykulazhenkov `drain.Helper` structure that is used in drain API, receives `PodSelector` as a single string. https://pkg.go.dev/k8s.io/kubectl/pkg/drain#Helper

Ok, thanks for the clarification, then it is totally fine to keep it as a string

Reply:
my main concern here is that its not clear what is the format of PodSelector.
if its well defined then we should use a structured type (and convert to string later) so it will be less error prone.

should it be of type[1] ?
or does the drain API only supports matching on labels ?

also i see DrainSpec.PodSelector is not being used in Drain manager. is that on purpose or just something left out ?


[1] https://github.com/k8snetworkplumbingwg/multi-networkpolicy/blob/220baf5d60c179716ed02d44cb0e646c76c1b605/vendor/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L1117

Reply:
ack, can you add a comment on it ? so its clear what is the expected content of this field

Reply:
See the updated description